src/date-time.ts

```diff
@@ -0,0 +1,553 @@
+
+        return new DateTime({
+            value: dateTimeString,
+            zone
+        });
+    }
+
+    public static min(dateTime1: DateTime, dateTime2: DateTime): DateTime
+    {
+        given(dateTime1, "dateTime1").ensureHasValue().ensureIsType(DateTime);
+        given(dateTime2, "dateTime2").ensureHasValue().ensureIsType(DateTime);
+
+        if (dateTime1.valueOf() < dateTime2.valueOf())
+            return dateTime1;
+
+        return dateTime2;
+    }
+
+    public static max(dateTime1: DateTime, dateTime2: DateTime): DateTime
+    {
+        given(dateTime1, "dateTime1").ensureHasValue().ensureIsType(DateTime);
+        given(dateTime2, "dateTime2").ensureHasValue().ensureIsType(DateTime);
+
+        if (dateTime1.valueOf() > dateTime2.valueOf())
+            return dateTime1;
+
+        return dateTime2;
+    }
+
+    public static validateDateTimeFormat(value: string): boolean
+    {
+        // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
+        if (value == null || value.isEmptyOrWhiteSpace())
+            return false;
+
+        return LuxonDateTime.fromFormat(value, DateTime._format).isValid;
+    }
+
+    public static validateDateFormat(value: string): boolean
+    {
+        // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
+        if (value == null || value.isEmptyOrWhiteSpace())
+            return false;
+
+        return LuxonDateTime.fromFormat(value, "yyyy-MM-dd").isValid;
+    }
+
+    public static validateTimeFormat(value: string): boolean
+    {
+        // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
+        if (value == null || value.isEmptyOrWhiteSpace())
+            return false;
+
+        return LuxonDateTime.fromFormat(value, "HH:mm").isValid;
+    }
+
+    public static validateTimeZone(zone: string): boolean
+    {
+        // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
+        if (zone == null || zone.isEmptyOrWhiteSpace())
+            return false;
+
+        try
+        {
+            DateTime._validateZone(zone);
+        }
+        catch
+        {
+            return false;
+        }
+
+        return LuxonDateTime.now().setZone(zone).isValid;
+    }
+
+
+    private static _validateZone(zone: string): void
+    {
+        if (zone.toLowerCase() === "utc")
+            return;
+
+        given(zone, "zone")
+            .ensureWhen(
+                zone.toLowerCase() === "local",
+                _ => false,
+                "should not use local zone")
+            .ensureWhen(
+                IANAZone.isValidSpecifier(zone),
+                t => IANAZone.isValidZone(t),
+                "Invalid IANA zone")
+            .ensureWhen(
+                zone.toLowerCase().startsWith("utc+"),
+                t =>
+                {
+                    // range is +00:00 to +14:00
+                    const offset = t.split("+").takeLast();
+
+                    if (offset.contains(":"))
```

Comment:
You can normalize it before, rather than doing the `:` if/else
normalize as in it's what's passed is +11 => +11:00